### PR TITLE
Fix select default channel

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This PR https://github.com/mattermost/mattermost-mobile/pull/2674 introduced a bug when selecting the default channel as it was using the selector `getChannelByName` instead of the utility of the same name thus it selected the first available channel that matched the name regardless of the team that it actually belonged.

### CODE REVIEW

File: app/actions/views/channel.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ import {
     getCurrentChannelId,
     getMyChannelMember,
     getRedirectChannelNameForTeam,
-    getChannelByName as getChannelByNameSelector,
+    getChannelsNameMapInCurrentTeam,
```
Comment (jespino) [inline app/actions/views/channel.js]: It should be `getChannelsNameMapInTeam` not `getChannelsNameMapInCurrentTeam`

File: app/actions/views/channel.js [hunk 2]
```diff
@@ -340,14 +341,17 @@ export function selectDefaultChannel(teamId) {
...
+
         let channelId;
         if (channel) {
             channelId = channel.id;
         } else {
             // Handle case when the default channel cannot be found
             // so we need to get the first available channel of the team
-            const channelsInTeam = Object.values(state.entities.channels).filter((c) => c.team_id === teamId);
+            const {channels} = state.entities.channels;
+            channelsInTeam = Object.values(channels).filter((c) => c.team_id === teamId);
```
Comment (jespino) [inline app/actions/views/channel.js]: Probably we can use directly the "channelsInTeam" created before.

**Comment by enahum** [comment]
@jespino changes made

**Comment by enahum** [comment]
Just checked that this will need a mattermost-redux PR

**Pull request merged**